troubleshooting/known_issues/SBPS_boot_fail.md

```diff
@@ -0,0 +1,150 @@
+Nov 12 19:20:05 ncn-w002 systemd[1]: sbps-marshal.service: Failed with result 'exit-code'.
+Nov 12 19:20:05 ncn-w002 systemd[1]: sbps-marshal.service: Consumed 1.394s CPU time.
+```
+
+The complete log for the SBPS Marshal agent `systemd` service can be obtained from `journalctl` log as below:
+
+```bash
+    # journalctl -u sbps-marshal.service
+```
+
+## Workaround Description
+
+Step 1: Identify the manifest(s) which are missing the `etag` field
+
+This can be fetched from worker node as `s3:/boot-images` is mounted onto the worker node. The following command will
+search the mounted directory for manifests that are missing the `etag` field:
+
+```bash
+for f in /var/lib/cps-local/boot-images/*/manifest.json; do \
+    echo $f etag=$(cat $f | jq '.artifacts[] | select(.type == "application/vnd.cray.image.rootfs.squashfs").link.etag' );  \
+done | grep etag=null
+```
+
+An example run of the command listing the problematic `manifest.json` is as follows:
+
+```bash
+# for f in /var/lib/cps-local/boot-images/*/manifest.json; do \
+     echo $f etag=$(cat $f | jq '.artifacts[] | select(.type == "application/vnd.cray.image.rootfs.squashfs").link.etag' );  \
+ done | grep etag=null
+/var/lib/cps-local/boot-images/3f89b76d-7578-4ac3-8891-0f8b4335b06d/manifest.json etag=null
+```
+
+Looking at the `manifest.json`:
+
+```bash
+ncn-w006:~ # cat /var/lib/cps-local/boot-images/3f89b76d-7578-4ac3-8891-0f8b4335b06d/manifest.json
+{
+   "artifacts" : [
+      {
+         "link" : {
+            "path" : "s3://boot-images/3f89b76d-7578-4ac3-8891-0f8b4335b06d/compute.squashfs",
+            "type" : "s3"
+         },
+         "md5" : "fa69a6665124629bfb1d74bfa3e4b7de",
+         "type" : "application/vnd.cray.image.rootfs.squashfs"
+      }
+   ],
+   "created" : "20240815093713",
+   "version" : "1.0"
+}
+```
+
+The "link" section has no `etag` field.
+
+Step 2: Update the manifest with `etag` and update the same in `s3`
+
+Example `manifest.json` update with `etag` obtained from `s3` and update the same in `s3`:
+
+```bash
+ncn-m001:~ # MANIFEST=$( cray ims images describe "3f89b76d-7578-4ac3-8891-0f8b4335b06d"  --format json | jq -r .link.path )
+ncn-m001:~ # echo $MANIFEST
+s3://boot-images/3f89b76d-7578-4ac3-8891-0f8b4335b06d/manifest.json
+
+ncn-m001:~ # cray artifacts get $( echo $MANIFEST | sed -e 's/^s3:\/\///g' -e 's/boot-images\//boot-images /g' ) manifest.json
+ncn-m001:~ # cat manifest.json
+{ "artifacts" : [ { "link" : { "path" : "s3://boot-images/3f89b76d-7578-4ac3-8891-0f8b4335b06d/compute.squashfs", "type" : "s3" }, "md5" : "fa69a6665124629bfb1d74bfa3e4b7de", "type" : "application/vnd.cray.image.rootfs.squashfs" } ], "created" : "20240815093713", "version" : "1.0" }
+ncn-m001:~ # cat manifest.json | jq -r '.artifacts[] | select(.type == "application/vnd.cray.image.rootfs.squashfs").link.path' | sed -e 's/^s3:\/\///g' -e 's/boot-images\//boot-images /g'

```

Comment:
```
sed -e 's/^s3:\/\///g' -e 's/boot-images\//boot-images /g'
```
This can be simplified to remove the backslashes by using a different character for the substitution separator:
```
sed -e 's;^s3://;;g' -e 's;boot-images/;boot-images ;g'
```